public/app/features/explore/spec/helper/interactions.ts

```diff
@@ -0,0 +1,28 @@
+import { selectors } from '@grafana/e2e-selectors';
+import userEvent from '@testing-library/user-event';
+import { fireEvent, screen, waitFor } from '@testing-library/react';
+
+export const changeDatasource = async (name: string) => {
+  const datasourcePicker = (await screen.findByLabelText(selectors.components.DataSourcePicker.container)).children[0];
+  fireEvent.keyDown(datasourcePicker, { keyCode: 40 });
+  const option = screen.getByText(name);
+  fireEvent.click(option);
+};
+
+export const inputQuery = (query: string) => {
+  const input = screen.getByRole('textbox', { name: 'query' });
+  userEvent.type(input, query);
+};
+
+export const runQuery = () => {
+  const button = screen.getByTestId(selectors.components.RefreshPicker.runButtonV2);
+  userEvent.click(button);
+};
+
+export const openQueryHistory = async () => {
+  const button = screen.getByRole('button', { name: 'Rich history button' });
+  userEvent.click(button);
+  await waitFor(() => {
+    expect(screen.getByText('The history is local to your browser and is not shared with others.')).toBeInTheDocument();
+  });
```

Comment:
you can just do

```suggestion
  expect(await screen.findByText('The history is local to your browser and is not shared with others.')).toBeInTheDocument();
```

(findBy* is a wrapper around waitFor and getBy*)